Library/Homebrew/hardware.rb

```diff
@@ -55,6 +55,8 @@ def self.cores_as_words
     when 1 then "single"
     when 2 then "dual"
     when 4 then "quad"
+    when 6 then "hexa"
```

Comment:
Amazon.com is giving me tons of results for “hexa-core”, but like all of these you have to spell it with the hyphen (like we do in our `brew config` output). FYI, https://en.wikipedia.org/wiki/Multi-core_processor is my primary source. I do agree that “six-core” seems to be much more widely used if Googling “six-core CPU” has any meaning, but “hex-core CPU” reads super-awkward to me and actually yields fewer results than “hexa-core CPU”, so I'm not quite sure which source to trust or whether I should just drop this PR.

On the bright side, this PR sparked a nice and useful discussion about physical vs. logical cores.

If there are really no good terms that are universally understood, maybe it's best to actually backtrack a little bit. How about we replace the existing logic with either of these three outputs?

- `single-core`
- `multi-core (N cores)`
- `multi-core (N physical, M logical cores)` for `N != M`